### PR TITLE
Add validation for SecretsManagerKmsKeyId

### PR DESCRIPTION
**Pull request by jonife**
*Description of changes:*
Added validation to SecretsManagerKmsKeyId 

*Description of how you validated changes:*
validation will ensure that SecretsManagerKmsKeyId is of type string

*Checklist:*

- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [x] Correct values
    - [x] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/eventsources/pull.py [hunk 1]
```diff
@@ -188,6 +188,13 @@ def _validate_filter_criteria(self):
         if list(self.FilterCriteria.keys()) not in [[], ["Filters"]]:
             raise InvalidEventException(self.relative_id, "FilterCriteria field has a wrong format")
 
+    def validate_secrets_manager_kms_key_id(self):
```
Comment (moelasmar) [inline samtranslator/model/eventsources/pull.py:191]: can you make it private method, and do the validation only if the SecretsManagerKmsKeyId is not None.

File: tests/translator/input/error_function_with_mq_kms_invalid_type.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Parameter:
```
Comment (moelasmar) [inline tests/translator/input/error_function_with_mq_kms_invalid_type.yaml:1]: can you add another test case to cover the self managed Kafka event.

**Review by moelasmar — CHANGES_REQUESTED** [review]

File: samtranslator/model/eventsources/pull.py (hunk 1 above)
Comment (jonife) [reply samtranslator/model/eventsources/pull.py:191]: the validation is called only when `SecretsManagerKmsKeyId` is provided. It is not a required property for an eventsource type

**Review by moelasmar — APPROVED** [review]

**Review by jfuss — COMMENTED** [review]
Can we add unit tests as well? The end to end tests are great but we can create a better coverage if we add unit tests and validate different input (lists, dict, int, etc).

File: tests/model/eventsources/test_mq_event_source.py [hunk 3]
```diff
@@ -40,3 +41,49 @@ def test_get_policy_statements(self):
             }
         ]
         self.assertEqual(policy_statements, expected_policy_document)
+
+    def test_must_validate_secrets_manager_kms_key_id(self):
+        self.mq_event_source.SourceAccessConfigurations = [{"Type": "BASIC_AUTH", "URI": "SECRET_URI"}]
+        self.mq_event_source.Broker = "BROKER_ARN"
+        self.mq_event_source.SecretsManagerKmsKeyId = ["1abc23d4-567f-8ab9-cde0-1fab234c5d67"]
```
Comment (jfuss) [inline tests/model/eventsources/test_mq_event_source.py:48]: Could we parameterize this to include anything that is not a string? We covered list here but would be good to add int, bool and dict to add move coverage.

**Review by jfuss — CHANGES_REQUESTED** [review]

**Review by jfuss — APPROVED** [review]
Thank you for going through additional iterations with adding more unit tests.

**Pull request merged**